src/case_encoder.h

```diff
@@ -179,34 +184,39 @@ class UpperCaseEncoder : public CaseEncoder {
     auto nrm_it = normalized->cbegin();
     auto n2o_it = norm_to_orig->cbegin();
 
-    while(std::regex_search(sig_it, signature_.cend(), m, e)) {
-      auto span = m[0];
-      size_t len = std::distance(sig_it, span.first);
-      normalized_temp.insert(normalized_temp.end(), nrm_it, nrm_it + len);
-      norm_to_orig_temp.insert(norm_to_orig_temp.end(), n2o_it, n2o_it + len);
-
-      sig_it += len; 
-      nrm_it += len;
-      n2o_it += len;
-      normalized_temp.push_back(cAllUppercase);
-      norm_to_orig_temp.push_back(*n2o_it);
+    try {
+      while(std::regex_search(sig_it, signature_.cend(), m, e)) {
+        auto span = m[0];
+        size_t len = std::distance(sig_it, span.first);
+        normalized_temp.insert(normalized_temp.end(), nrm_it, nrm_it + len);
+        norm_to_orig_temp.insert(norm_to_orig_temp.end(), n2o_it, n2o_it + len);
+
+        sig_it += len; 
+        nrm_it += len;
+        n2o_it += len;
+        normalized_temp.push_back(cAllUppercase);
+        norm_to_orig_temp.push_back(*n2o_it);
             
-      while(sig_it != span.second) {
-        if(*sig_it == cUppercase) {
-          sig_it++; 
-          nrm_it++;
-          n2o_it++;
+        while(sig_it != span.second) {
+          if(*sig_it == cUppercase) {
+            sig_it++; 
+            nrm_it++;
+            n2o_it++;
+          }
+          sig_it++;
+          normalized_temp.push_back(*nrm_it++);
+          norm_to_orig_temp.push_back(*n2o_it++);
         }
-        sig_it++;
-        normalized_temp.push_back(*nrm_it++);
-        norm_to_orig_temp.push_back(*n2o_it++);
-      }
-      if(sig_it != signature_.cend()) { 
-        if(*sig_it != cUppercase) {
-          normalized_temp.push_back(cLowercase);
-          norm_to_orig_temp.push_back(*n2o_it);
+        if(sig_it != signature_.cend()) { 
+          if(*sig_it != cUppercase) {
+            normalized_temp.push_back(cLowercase);
+            norm_to_orig_temp.push_back(*n2o_it);
+          }
         }
       }
+    } catch (std::regex_error&) {
+        LOG(WARNING) << "Regex error encountered in case encoding; rejecting sentence";
```

Comment:
How many of those messages can be seen so that they really shouldn't be ignored? If this error can be quite frequent and thus potentially harmful, maybe this could be made clearer in the warning message? If this is infrequent, feel free to ignore this comment.